lib/business/business-base.mjs

```diff
@@ -296,13 +296,13 @@ class BusinessBase {
         }
         if (isStandard) {
             if (!isUpdate) {
-                if (user.id) {
-                    values.CreatedByUserId = user.id;
+                if (user.userDetails.id) {
```

Comment:
You should not change structure of user object because this same framework code is being used in other applications too which will break functionality.

Instead, you should handle in your current project, and use existing code of dframework